src/components/measureEditor/MeasureEditor.tsx

```diff
@@ -217,11 +277,14 @@ const MeasureEditor = () => {
       // errorExceptions contains error data for the primary library,
       // aka the CQL loaded into the editor. Errors from included
       // libraries are available in data.annotations.errors, if needed.
-      const elmAnnotations = mapElmErrorsToAceAnnotations(allErrorsArray);
-      const errorMarkers = mapElmErrorsToAceMarkers(allErrorsArray);
-      setElmAnnotations(elmAnnotations);
+      let annotations = mapErrorsToAceAnnotations(allErrorsArray, "ELM");
+      annotations = [...annotations, ...vsacErrorsAnnotations];
+      let errorMarkers = mapErrorsToAceMarkers(allErrorsArray);
+      errorMarkers = [...errorMarkers, ...vsacErrorMarkers];
+
+      setElmAnnotations(annotations);
       setErrorMarkers(errorMarkers);
-      return data;
+      return translationResults;
```

Comment:
Only ELM translation results are being returned here (which seems like should correspond to what "data" was before). But should this be updated to return both elm translation errors as well as value set errors? Basically, if there are value set validation errors, should that set the cqlErrors flag to true? That flag drives whether or not the measure CQL is valid for test case execution, and may also drive measure bundling later. Or maybe this is something that might need to be addressed in a followup jira?

Reply:
That's a really good point, We definitely missed it while writing the story. Yes, do need to toggle the flag based on AntlrParse errors, translation errors, and VSAC errors. It looks like the value sets errors are also not accounted for that. So I will make the update for both Valueset errors and Codes

Reply:
Hey Nate, I was thinking of writing a new story for this update. I thought I could simply return translationResults ( which includes both translation and VSAC errors ) but the translationResults is not just used for the flag, but it is an elm representation that is stored in DB. So We have to make changes in such a way that we capture these 2 errors individually. that required vsac errors to be separated, but the way valueset is implemented that, it is dependent on translationResults which is not necessary.